### DISCUSSION TITLE
Attribute inaccessible for some reason

### DISCUSSION THREAD

**Discussion by StotreB** [ansys/pyaedt#4050] [Q&A] [answered]
Hello,

I am trying to write into a text file part of the setup of some simulations. They are performed using Q3D and I can access some setup data by accessing the attributes of my apps, when running:

dir(app)

 I can see an attribute of my app called excitations_by_type, which contains some information which I would like to store, however, when I run: 

getattr(app, "excitations_by_type")

I get the following error: {AttributeError}'Q3d' object has no attribute '_excitations'.

How did "excitations_by_time" became "_excitations" and why can't I access this attribute?

B.R.

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @StotreB ,

Not sure what you are trying to achieve, but excitations_by_type is a property of the design, you can directly access to the data:

aedtapp.excitations_by_type

or if you know the type of excitation:

aedtapp.excitations_by_type["Coil"]

If it does not work, maybe is because you are using an excitation that is not covered by PyAEDT yet. So please, share more information regarding the Q3D project (you can even attach it here), and the snipped code to reproduce the issue.
